strawberry/field.py

```diff
@@ -23,129 +23,289 @@
-
-        kwargs: Dict[str, Any] = {}
-
-        # kw_only was added to python 3.10 and it is required
-        if sys.version_info >= (3, 10):
-            kwargs["kw_only"] = False
+        is_basic_field: bool = False,
+        kwargs: dict = None,
+    ) -> "DataclassFieldSettings":
+        return cls(
+            default,
+            default_factory,
+            is_basic_field=is_basic_field,
+            kwargs=kwargs if kwargs else {},
+        )
 
-        super().__init__(
-            default=(default if default is not UNSET else dataclasses.MISSING),
+    def to_dataclass_field(self) -> dataclasses.Field:
+        kwargs = dataclasses.asdict(self)
+        kwargs.update(kwargs.pop("kwargs"))
+        # basic fields are fields with no provided resolver
+        default = dataclasses.MISSING
+        default_factory = dataclasses.MISSING
+        if not self.is_basic_field:
+            if self.default is not UNSET:
+                default = self.default
+            if self.default_factory is not UNSET:
+                default_factory = self.default_factory
+        return dataclasses.Field(
+            default=default,
             default_factory=(
                 # mypy is not able to understand that default factory
                 # is a callable so we do a type ignore
                 default_factory  # type: ignore
-                if default_factory is not UNSET
-                else dataclasses.MISSING
             ),
-            init=is_basic_field,
-            repr=is_basic_field,
-            compare=is_basic_field,
+            init=self.is_basic_field,
+            repr=self.is_basic_field,
+            compare=self.is_basic_field,
             hash=None,
             metadata={},
             **kwargs,
         )
 
-        self.graphql_name = graphql_name
-        if python_name is not None:
-            self.python_name = python_name
-
-        self.type_annotation = type_annotation
 
-        self.description: Optional[str] = description
-        self.origin = origin
+DEFAULT_FIELD_SETTINGS = DataclassFieldSettings()
 
-        self._base_resolver: Optional[StrawberryResolver] = None
-        if base_resolver is not None:
-            self.base_resolver = base_resolver
 
-        # Note: StrawberryField.default is the same as
-        # StrawberryField.default_value except that `.default` uses
-        # `dataclasses.MISSING` to represent an "undefined" value and
-        # `.default_value` uses `UNSET`
-        self.default_value = default
-        if callable(default_factory):
-            try:
-                self.default_value = default_factory()
-            except TypeError as exc:
-                raise InvalidDefaultFactoryError() from exc
-
-        self.is_subscription = is_subscription
+@dataclasses.dataclass(frozen=True)
+class StrawberryField:
+    python_name: Optional[str]
+    """
+    name to be used by python, i.e in the generated __init__
+    """
```

Comment:
nit: I don't we need all these doc strings, I think they make the code less readable and don't provide a lot of value 😊

Reply:
This can make it easier to generate docs later on though...